tt_eager/tt_dnn/op_library/auto_format.cpp

```diff
@@ -94,14 +93,14 @@ Tensor AutoFormat::format_input_tensor(
             }
         } else if (!convert_layout && pad_input) {
             if (formatted_input.get_layout() == Layout::ROW_MAJOR || formatted_input.get_layout() == Layout::TILE) {
-                return ttnn::pad(0, (const ttnn::Tensor) formatted_input, ttnn::Shape(padded_shape), ttnn::Shape({0, 0, 0, 0}), pad_value, false, mem_config);
+                return ttnn::pad(0, (const ttnn::Tensor) formatted_input, padded_shape.to_vector(), std::vector<uint32_t>({0, 0, 0, 0}), pad_value, false, mem_config);
```

Comment:
This code is slower. One is based on std::array, the other is heap based dynamically allocated vector.

Reply:
@ayerofieiev-tt  I want to replace all vectors with something like small_vector in future.
So allocations will be solved all at once.
If boost is allowed there is a good candidate https://www.boost.org/doc/libs/1_65_1/doc/html/container/non_standard_containers.html#container.non_standard_containers.small_vector